pkg/metadata/scheduler.go

```diff
@@ -51,6 +52,9 @@ func (c *Scheduler) Stop() {
 	for _, h := range c.healthHandles {
 		h.Deregister()
 	}
+	for _, stopChannel := range c.stopChannels {
+		stopChannel <- struct{}{}
```

Comment:
Nits: I think you can close the channel here instead of sending an empty struct. Then you won't need a buffered channel here

```stopChan := make(chan struct{}, 1)```
https://github.com/DataDog/datadog-agent/pull/3448/files#diff-f6baa386e7bcc8bf9d64dc79cb19be42R69

Reply:
Thanks, updated

Reply:
Well, just a nitpick here but given that reading from a closed channel is non-blocking it is likely but not *certain* that these go routines have exited when `Stop()` returns. They may exit on the next tick or later. Using buffered channels would be the way to make sure of that.